src/coreclr/vm/syncblk.inl

```diff
@@ -683,7 +687,7 @@ FORCEINLINE AwareLock::LeaveHelperAction AwareLock::LeaveHelper(Thread* pCurThre
         MODE_ANY;
     } CONTRACTL_END;
 
-    if (m_HoldingThread != pCurThread)
+    if (m_HoldingThreadId != pCurThread->GetThreadId())
```

Comment:
Since the enter path is still using the thread pointer, maybe the exit path here could also use the thread pointer for now just for minor consistency. I assume the thread pointer will be removed separately, at which point all of the checks would switch to using the thread ID anyway.